packages/cli/lib/update-index.js

```diff
@@ -0,0 +1,23 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+const path = require('path');
+const util = require('util');
+const fs = require('fs');
+const appendFileAsync = util.promisify(fs.appendFile);
```

Comment:
Why not use this instead? https://nodejs.org/api/fs.html#fs_fs_appendfilesync_path_data_options

Reply:
`appendFileSync` is a blocking / synchronous operation. `async / await` is a promise under the hood even though it appears to be synchronous when written.

From https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9
> Async/await makes asynchronous code look and behave a little more like synchronous code. This is where all its power lies.

In a generator you can call `updateIndex` without an `await` ... and have more code below -- and that additional code and continue running without being blocked essentially. 

Here's a Node.js article on blocking / non-blocking. https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/